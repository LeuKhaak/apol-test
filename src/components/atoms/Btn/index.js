import React from "react";
import * as styles from "./styles.module.scss";
import cn from "classnames";

const Btn = ({ btnText, btnStyle, type, onClick, arg, arg2, id }) => {
  const classBtn = cn(styles.btn, styles[btnStyle], styles[type]);

  const onClickFunc = () => {
    if (onClick) onClick(arg, arg2);
  };

  return (
    <button
      className={classBtn}
      id={id}
      onClick={() => {
        onClickFunc();
      }}
    >
      <span className={styles.linkText}>{btnText}</span>
    </button>
  );
};

export default Btn;
