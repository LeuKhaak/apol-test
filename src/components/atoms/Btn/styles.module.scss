@import "../../../assets/styles/styles.scss";

.btn {
  cursor: pointer;
  height: 2rem;
  padding: 0.5rem;
  border-radius: 0.25rem;
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
}
.nav {
  background: $gold;
}

.edite,
.delete,
.submit,
.add,
.upload,
.export {
  color: $white;
  background-color: rgb(201, 40, 40);
}

.edite,
.submit {
  background-color: rgb(43, 102, 180);
  margin-right: 0.5rem;
}

.upload {
  background-color: rgb(77, 194, 96);
  margin-bottom: 0.5rem;
}

.export {
  background-color: rgb(77, 194, 96);
}

.add {
  background-color: rgb(71, 227, 233);
  margin-left: 55rem;
}

//.toggle {
//  padding: 0;
//  .icon {
//    cursor: pointer;
//    margin-right: 0;
//    width: 1.5rem;
//  }
//}
//.pageNumber {
//  @include pageButton;
//  padding: 0;
//}

//.pageNumberChecked {
//  @include pageButton;
//  padding: 0;
//  color: $gold;
//  background:
//  box-shadow: 0 0.25rem 0.5rem $shadowDark;
//  border-radius: 0.25rem;
//}

//.buttonNext {
//  @include pageButton;
//  transform: rotateY(180deg);
//}
