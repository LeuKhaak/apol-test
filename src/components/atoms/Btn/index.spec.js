import React from "react";
import { shallow } from "enzyme";
import Btn from ".";

let component = null;
const onClick = jest.fn();
const setComponent = (props) => shallow(<Btn {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("Btn testing:", () => {
  test("is Btn contain text", () => {
    component = setComponent({
      btnText: "testText",
      btnStyle: "testStyle",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn contain nothing", () => {
    expect(component).toMatchSnapshot();
  });

  test("is Btn not displayed", () => {
    component = setComponent({
      btnText: "testText",
      disp: "none",
    });
    expect(component).toMatchSnapshot();
  });

  test("check onClickFunc", () => {
    component = setComponent({
      onClick: onClick,
      btnText: "testText",
    });
    component.find("button").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
  test("check onClickFunc no called", () => {
    component = setComponent({
      btnText: "testText",
    });
    component.find("button").simulate("click");
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  test("is Btn has className", () => {
    component = setComponent({
      btnText: "testText",
      btnStyle: "buttonNext",
    });
    expect(component.find("buttonNext")).toBeTruthy();
  });
});
