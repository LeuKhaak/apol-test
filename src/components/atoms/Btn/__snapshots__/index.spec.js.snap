// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn testing: is Btn contain nothing 1`] = `
<button
  className=""
  onClick={[Function]}
>
  <span />
</button>
`;

exports[`Btn testing: is Btn contain text 1`] = `
<button
  className=""
  onClick={[Function]}
>
  <span>
    testText
  </span>
</button>
`;

exports[`Btn testing: is Btn not displayed 1`] = `
<button
  className=""
  onClick={[Function]}
>
  <span>
    testText
  </span>
</button>
`;
